eoc_journal/templates/eoc_journal.html

```diff
@@ -4,30 +4,91 @@
-            <p>{{ q.answer }}</p>
-          </div>
-        {% endfor %}
-      </section>
-      {% endfor %}
+  <div class="course-progress">
+    <div class="title">
+      <h2>Progress</h2>
+      <p>
+      You are <span data-progress-name="user">{{ progress.user }}</span>% of the way through the course based on
+      completion of modules within lessons. The cohort average is
+      <span data-progress-name="cohort">{{progress.cohort_average}}</span>%.
+      </p>
+    </div>
   </div>
-  {% endif %}
-
-  <p>
-    {% if key_takeaways_pdf_url %}
-    <a class="key-takeaways-link" target="_blank" href="{{ key_takeaways_pdf_url }}">
-      <span class="fa fa-file-text-o" aria-hidden="true"></span>
-      {% trans "Key Takeaways" %}
-    </a>
-    {% else %}
-      {% trans "Key Takeaways PDF not available at this time." %}
-    {% endif %}
-  </p>
+
+  <div class="course-engagement">
+    <div class="title">
+      <h2>{% trans "Engagement" %}</h2>
+
+      <p>
+      {% trans "Your current engagement score is "%} <span data-engagement-name="user">{{engagement.user_score}}</span>
+      {% trans "based on your engagement interactions with other participants in your cohort in the" %}
+      <a href="/courses/{{course.id}}/discussion">{% trans "discussion" %}</a>
+      {% trans "forums. The cohort average is" %} <span data-engagement-name="cohort">{{engagement.cohort_score}}</span>
+      </p>
+    </div>
+
+    <table>
+      <thead>
+        <tr>
+          <th>Measured engagement</th>
+          <th>Stats</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        <tr>
+          <td>{% trans "Number of New Posts you created" %}</td>
+          <td data-point-name="new_posts">{{ engagement.new_posts }}</td>
+        </tr>
+
+        <tr>
+          <td>{% trans "Number of Replies you made to other posts" %}</td>
+          <td data-point-name="replies">{{ engagement.replies }}</td>
+        </tr>
+
+        <tr>
+          <td>{% trans "Number of Comments Generated by your posts" %}</td>
+          <td data-point-name="comments_generated">{{ engagement.comments_generated }}</td>
+        </tr>
+
+        <tr>
+          <td>{% trans "Number of Upvotes your posts have received" %}</td>
+          <td data-point-name="upvotes">{{ engagement.upvotes }}</td>
+        </tr>
+
+        <tr>
+          <td>{% trans "Number of your posts being Followed" %}</td>
+          <td data-point-name="posts_followed">{{ engagement.posts_followed }}</td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</div>
+
+{% if answer_sections %}
+<div class="eoc-selected-answers">
+  {% for section in answer_sections %}
+  <section class="eoc-selected-answers-section">
+    <h2>{{ section.name }}</h2>
+    {% for q in section.questions %}
+    <div class="eoc-selected-answer">
+      <h3>{{ q.question }}</h3>
+      <p>{{ q.answer }}</p>
+    </div>
+    {% endfor %}
+  </section>
+  {% endfor %}
+</div>
+{% endif %}
+
+<p>
```

Comment:
Nit: while you're touching this code, could you move the `<p>` tag inside the `{% if %} ... {% endif %}` block, please?